lib/xcov/model/report.rb

```diff
@@ -66,7 +66,14 @@ def self.map dictionary
     def self.filter_targets targets
       filtered_targets = Array.new(targets)
       filtered_targets = filtered_targets.select { |target| !target["name"].include?(".xctest") } if !Xcov.config[:include_test_targets]
-      filtered_targets = filtered_targets.select { |target| !self.excluded_targets.include?(target["name"])}
+
+      if Xcov.config[:exclude_targets]
+        filtered_targets = filtered_targets.select { |target| !self.excluded_targets.include?(target["name"])}
+      end
+
+      if Xcov.config[:include_targets]
+        filtered_targets = filtered_targets.select { |target| self.included_targets.include?(target["name"])}
+      end
```

Comment:
I would move this statement to the beginning of the method and then return. Because if we specify the `include_targets` we shouldn't bother about excluded ones, am I right?